redis.conf

```diff
@@ -1829,6 +1829,19 @@ activerehashing yes
 # Instead there is a default limit for pubsub and replica clients, since
 # subscribers and replicas receive data in a push fashion.
 #
+# Replica clients output buffer limit should be more than replication backlog
+# size, since master may copy almost whole replication backlog to replica's
+# output buffer if it is able to perform a partial resynchronization, but we
+# will close replica client and resynchronization is failed if output buffer
+# limit is reached. This situation might continue until replication backlog
+# start offset is out of the PSYNC offset, and causes full synchronization.
+# So If you set the replica output buffer limit less than replication backlog
+# size, to avoid above bad situation, if the replica output buffer size is not
+# more than replication backlog size but already exceeds output buffer hard
+# limit, we don't think hard limit is reached and don't close replica client.
+# This method doesn't cost more memory since replication backlog and replicas
+# share one global replication buffer.
```

Comment:
I think this is too detailed, IMHO it should be enough to say:

```suggestion
# Note that it doesn't make sense to set the replica clients output buffer
# limit lower than the repl-backlog-size config (partial sync will succeed
# and then replica will get disconnected).
# Such a configuration is ignored (the size of repl-backlog-size will be used).
# This doesn't have memory consumption implications since the replica client
# will share the backlog buffers memory.
```